src/Razor/test/Microsoft.VisualStudio.Editor.Razor.Test/TelemetryReporterTests.cs

```diff
@@ -322,4 +323,107 @@ public void TrackLspRequest()
                 Assert.Equal(correlationId, correlationProperty.Value);
             });
     }
+
+    [Fact]
+    public void ReportFault_OperationCanceledExceptionWithoutInnerException_SkipsFaultReport()
+    {
+        // Arrange
+        var reporter = new TestTelemetryReporter(new RazorLoggerFactory([]));
+        var exception = new OperationCanceledException("OCE", innerException: null);
+
+        // Act
+        reporter.ReportFault(exception, "Test message");
+
+        // Assert
+        Assert.Empty(reporter.Events);
+    }
+
+    [Fact]
+    public void ReportFault_TaskCanceledExceptionWithoutInnerException_SkipsFaultReport()
+    {
+        // Arrange
+        var reporter = new TestTelemetryReporter(new RazorLoggerFactory([]));
+        var exception = new TaskCanceledException("TCE", innerException: null);
+
+        // Act
+        reporter.ReportFault(exception, "Test message");
+
+        // Assert
+        Assert.Empty(reporter.Events);
+    }
+
+    [Theory]
+    [InlineData(3)]
+    public void ReportFault_InnerExceptionOfOCEIsAnotherOCE_SkipsFaultReport(int depth)
+    {
+        // Arrange
+        var reporter = new TestTelemetryReporter(new RazorLoggerFactory([]));
+        var innerMostException = new OperationCanceledException();
+        var exception = CreateNestedException(typeof(OperationCanceledException), depth, innerMostException);
+
+        // Act
+        reporter.ReportFault(exception, "Test message");
+
+        // Assert
+        Assert.Empty(reporter.Events);
+    }
+
+    [Theory]
+    [InlineData(3)]
+    public void ReportFault_InnerExceptionOfOCEIsNotAnOCE_ReportsFault(int depth)
+    {
+        // Arrange
+        var reporter = new TestTelemetryReporter(new RazorLoggerFactory([]));
+        var innerMostException = new Exception();
+        var exception = CreateNestedException(typeof(OperationCanceledException), depth, innerMostException);
+
+        // Act
+        reporter.ReportFault(exception, "Test message");
+
+        // Assert
+        Assert.NotEmpty(reporter.Events);
+    }
+
+    [Theory]
+    [InlineData(3)]
+    public void ReportFault_InnerMostExceptionIsOperationCanceledException_SkipsFaultReport(int depth)
```

Comment:
This test looks to be identical to `ReportFault_InnerExceptionOfOCEIsAnotherOCE_SkipsFaultReport` above